src/models/flight_control/FGLinearActuator.cpp

```diff
@@ -0,0 +1,257 @@
+/* %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+ * 
+ * Module:       FGLinearActuator.cpp
+ * Author:       Adriano Bassignana
+ * Date started: 2019-01-03
+ * 
+ * ------------- Copyright (C) 2000 -------------
+ * 
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place - Suite 330, Boston, MA  02111-1307, USA.
+ * 
+ * Further information about the GNU Lesser General Public License can also be found on
+ * the world wide web at http://www.gnu.org.
+ * 
+ * FUNCTIONAL DESCRIPTION
+ * --------------------------------------------------------------------------------
+ * 
+ * HISTORY
+ * --------------------------------------------------------------------------------
+ * 
+ * %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+ * COMMENTS, REFERENCES,  and NOTES
+ * %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+ * 
+ * %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+ * INCLUDES
+ * %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
+
+#include <stdlib.h>
+
+#include "FGLinearActuator.h"
+#include "input_output/FGXMLElement.h"
+#include "math/FGParameterValue.h"
+
+#include "models/FGFCS.h"
+
+using namespace std;
+
+namespace JSBSim {
+    
+    /* %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+     * CLASS IMPLEMENTATION
+     * %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
+    
+    FGLinear_Actuator::FGLinear_Actuator(FGFCS* fcs, Element* element) : FGFCSComponent(fcs, element)
+    {
+        
+        if (element->FindElement("set")) {
+            isSetProperty = true;
+            setProperty = PropertyManager->CreatePropertyObject<double>(element->FindElementValue("set"));
```

Comment:
Today I looked at the problem you asked me, but I could not find a way to evaluate with the "if" condition so as to understand that if setProperty and resetProperty were defined or not. Unfortunately, this object has seen that it is only used for FGWaypoint and FGPropulsion and in both cases, if the parameter is not defined, there is an error condition with the end of the program. In my case the non-definition of set or reset is a precise information that indicates that set and reset must operate in standard mode (ie set = 1, reset = 0). In this case the execution speed of a test on static boolean variable is much quicker to execute because the variable isSetProperty or isResetProperty are normally in stak and therefore they are almost always in the processor's pipe. If you find me an alternative solution that is just as quick and does not produce a run-time error I would be grateful.

Reply:
Change the `setProperty` type to `FGPropertyNode_ptr` and check for nullity
```diff
-        if (element->FindElement("set")) {
-            isSetProperty = true;
-            setProperty = PropertyManager->CreatePropertyObject<double>(element->FindElementValue("set"));
-        }
+        Element* set_el = element->FindElement("set");
+        if (set_el) {
+            string propName = set_el->GetDataLine();
+            setProperty = PropertyManager->GetNode(propName);
+        }
+        else
+            setProperty = nullptr;
```

Reply:
> [...snip...] static boolean variable is much quicker to execute because the variable isSetProperty or isResetProperty are normally in stack [...snip...]

This code is in the constructor so it will be executed once. Optimization is irrelevant here.